### PR TITLE
dataset duck typing for dataloader

### PR DESCRIPTION
**Pull request by jperez999**
removing knowledge of dataset and everything else outside the Tags and dispatch from dataloader package.

### CODE REVIEW

**Comment by rjzamora** [comment]
Not that I have any problem with this kind of change, but is there a motivating issue or explanation of the rational here?

**Comment by karlhigley** [comment]
@rjzamora I think this is related to wanting to move the dataloaders into the new models library without that library needing to depend on NVTabular. From talking to @jperez999 earlier, it sounds like allowing the dataloaders to operate on a plain ddf makes it easier to cleanly separate datasets (in NVT) from dataloaders (in the models library in a way that lets them be used either together or separately.

**Comment by rjzamora** [comment]
Sounds good @karlhigley !  This does make sense to me. However, since I occasionally find myself tracking down performance regressions in NVTabular, I like there to be a linked issue or motivation described in any PR touching the IO interface.  Your last comment seems sufficient to me :)

**Comment by jperez999** [comment]
rerun tests

File: tests/unit/loader/test_dataloader_backend.py [hunk 1]
```diff
@@ -122,11 +122,22 @@ def test_dataloader_empty_error(datasets, engine, batch_size):
     )
 
 
+class DatasetStub:
+    def __init__(self, dataset, schema):
+        self.data = dataset
+        self.schema = schema
+
+
```
Comment (karlhigley) [inline tests/unit/loader/test_dataloader_backend.py:130]: It doesn't look like this actually gets used in this file:
```suggestion
```

File: nvtabular/loader/torch.py [hunk 2]
```diff
@@ -87,6 +87,8 @@ def __init__(
         sparse_max=None,
         sparse_as_dense=False,
     ):
+
+        # dataset = nvt.dispatch.validate_ds(dataset)
```
Comment (karlhigley) [inline nvtabular/loader/torch.py:91]: Stray comment here:
```suggestion

```

File: nvtabular/loader/tensorflow.py [hunk 3]
```diff
@@ -33,12 +32,21 @@
...
-def _validate_dataset(paths_or_dataset, batch_size, buffer_size, engine, reader_kwargs):
+dd_engine = {
+    "parquet": dd.read_parquet,
+    "csv": dd.read_csv,
+    "df": dd.DataFrame,
+}
+
+
+def _validate_dataset(
+    paths_or_dataset, batch_size, buffer_size, engine, reader_kwargs, dataclass=None
```
Comment (karlhigley) [inline nvtabular/loader/tensorflow.py:43]: I think this extra parameter came from an intermediate version of the code, but doesn't look like it gets used here:
```suggestion
    paths_or_dataset, batch_size, buffer_size, engine, reader_kwargs
```

**Review by karlhigley — APPROVED** [review]

**Comment by jperez999** [comment]
rerun tests

**Pull request merged**